traffic_monitor/cache/stats_types.go

```diff
@@ -105,7 +105,7 @@ type StatsDecoder struct {
 // whatever miscellaneous data was in the payload but not represented by
 // the properties of a Statistics object, so that it can be used in later
 // calculations if necessary.
-type StatisticsParser func(string, io.Reader) (Statistics, map[string]interface{}, error)
+type StatisticsParser func(string, io.Reader, interface{}) (Statistics, map[string]interface{}, error)
```

Comment:
It looks like this is always typecast to `*poller.HTTPPollCtx` type. Any reason we can't skip the typecast and make this `*poller.HTTPPollCtx` directly (though I understand, separately, that `handler.Handler` uses `interface{}` type to avoid an import cycle)?

Reply:
I left it as interface{} since this is the generic definition of a parser, so it gets used by any potential stats parser plugins. So specifying it as httppoll ties peoples hands as to what a plugin would be able to accept. So I think its best to leave it as generic as possible